### PR TITLE
MGDAPI-3192

### PR DESCRIPTION
**Pull request by laurafitzgerald**
Improvements to how the serviceUpdates are updated in the cro strategy config map
to prevent the config map flipping alphabetically causing the cro reconciler to go
in progress incorrectly.

Also some clean up on imports and use of strings from cro packages
Removal of some unused cro functions

# Issue link
https://issues.redhat.com/browse/MGDAPI-3192

# What
See above

# Verification steps

Using CCS Cluster

For quick local verification 
Run `INSTALLATION_TYPE=managed-api USE_CLUSTER_STORAGE=false make cluster/prepare/local`
And  `INSTALLATION_TYPE=managed-api USE_CLUSTER_STORAGE=false make code/run`

Verify that the config map is not flipping back and forth as it described in the ticket. 

For full upgrade verification which is recommended follow steps in -> https://github.com/integr8ly/integreatly-operator/pull/2291

### CODE REVIEW

File: pkg/resources/awsStrategyHelper.go [hunk 1]
```diff
@@ -1,738 +0,0 @@
...
-
-const (
-	configMapName    = "cloud-resources-aws-strategies"
-	PostgresStratKey = "postgres"
-	RedisStratKey    = "redis"
-)
-
-// At the moment this function only allows for overriding the createStrategy. It's unlikely that delete or region will need to be
-// overridden. In the case that they do, the function can be extended.
-func CreatePostgresTierFromProduction(ctx context.Context, client k8sclient.Client, ns string, useClusterStorage bool, tier string, createStrategy map[string]interface{}, logger l.Logger) error {
```
Comment (laurafitzgerald) [inline pkg/resources/awsStrategyHelper.go:34]: These are unused inside the project. there are some more specific helpers inside the CRO project which should probably be used instead. If necessary I can move this change to another pr as it's not essential for the bug fix.

**Comment by MStokluska** [comment]
@laurafitzgerald can you `make code/fix` on it please, apart from that, all looks good!

**Comment by MStokluska** [comment]
/lgtm
All looks good to me, I've verified it on a cluster with fresh local install and olm upgrade scenario and it works as expected. RHMI CR isn't being refreshed. Code looks fine too.
Good job.

**Pull request merged**